In [ ]:
!pip install -q pyomo

let's suppose that Jen Industries produces  $  x_1 kgs of product P and x_2 kgs of product Q $ 
$\min \ 1200x_1+2580x_2\ \\ \text{s.t.} \ x_1+x_2 \geq 145 \\ 2.18x_1+\frac{13}{12}x_2 \leq 980, \ x_1 \geq 105 , \ x_2 \geq 125 $


In [ ]:
from pyomo.environ import *

In [ ]:
# Create a model3
model3=ConcreteModel()

In [ ]:
#declare decision Variables
model3.x1= Var(domain= Reals)
model3.x2=Var(domain=Reals)


In [ ]:
#declare objective
model3.Cost = Objective(expr =1200*model3.x1+2580*model3.x2 , sense=minimize)

In [ ]:
#declare costraints
model3.constr1= Constraint(expr=model3.x1+model3.x2>=145)
model3.constr2=Constraint(expr=2.18*model3.x1+1.08*model3.x2<=980)
model3.constr3=Constraint(expr=model3.x1>=105)
model3.constr4=Constraint(expr=model3.x2>=125)

In [ ]:
model3.pprint()


2 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals

1 Objective Declarations
    Cost : Size=1, Index=None, Active=True
        Key  : Active : Sense    : Expression
        None :   True : minimize : 1200*x1 + 2580*x2

4 Constraint Declarations
    constr1 : Size=1, Index=None, Active=True
        Key  : Lower : Body    : Upper : Active
        None : 145.0 : x1 + x2 :  +Inf :   True
    constr2 : Size=1, Index=None, Active=True
        Key  : Lower : Body              : Upper : Active
        None :  -Inf : 2.18*x1 + 1.08*x2 : 980.0 :   True
    constr3 : Size=1, Index=None, Active=True
        Key  : Lower : Body : Upper : Active
        None : 105.0 :   x1 :  +Inf :   True
    constr4 : Size=1, Index=None, Active=T

In [ ]:
!apt-get install -y -qq glpk-utils

In [ ]:
SolverFactory('glpk', executable='/usr/bin/glpsol').solve(model3).write()

# ==========================================================
# = Solver Results                                         =
# ==========================================================
# ----------------------------------------------------------
#   Problem Information
# ----------------------------------------------------------
Problem: 
- Name: unknown
  Lower bound: 448500.0
  Upper bound: 448500.0
  Number of objectives: 1
  Number of constraints: 5
  Number of variables: 3
  Number of nonzeros: 7
  Sense: minimize
# ----------------------------------------------------------
#   Solver Information
# ----------------------------------------------------------
Solver: 
- Status: ok
  Termination condition: optimal
  Statistics: 
    Branch and bound: 
      Number of bounded subproblems: 0
      Number of created subproblems: 0
  Error rc: 0
  Time: 0.011070966720581055
# ----------------------------------------------------------
#   Solution Information
# ------------------------------

In [ ]:
#display solution
print('\nCost=',model3.Cost())
print('decision Variables')
print('x1=', model3.x1())
print('x2=',model3.x2())
print('Constraints')
print('constraint 1=',model3.constr1())
print('constraint2=',model3.constr2())
print('constraint3=',model3.constr3())
print('constraint4=',model3.constr4())



Cost= 448500.0
decision Variables
x1= 105.0
x2= 125.0
Constraints
constraint 1= 230.0
constraint2= 363.9
constraint3= 105.0
constraint4= 125.0



1ans.  Jen Industries produces x_1kgs of product P, \\
   and produces y kgs of product Q, 
2 ans.  optimal cost are: 448500.0 \\
   value of decision variable:
   x1 =  105.0,
x2 =  125.0 \\
   Activeness of constraints: \\
   constraint 1  not active because LHS= x1+x2 = 105+125 =230  is not equal to RHS. \\
   Constraint 2  not active because LHS = 4371  is not equal RHS \\
   constraint 3  active  LHS = 105  is equal to RHS \\
   constraint 4 is active  LHS = 125 because is equal to RHS \\

In [33]:
!pip install -q pyomo

$ "x_1 \geq 147 ", "x _2\geq 222"$
now the new changed Linear programming problem is as follows:
$\min \ 1200x_1+2580_2 \\ \text{s.t.} \ x_1+x_2\geq 145 \\ 2.18x+\frac{13}{12}x_2 \leq 980, \ x_1\geq 147 , \ x_2\geq 222 $

In [34]:
from pyomo.environ import *

In [37]:
# Create a model31
model31=ConcreteModel()

In [38]:
#declare decision Variables
model31.x1= Var(domain= Reals)
model31.x2=Var(domain=Reals)


In [40]:
#declare objective
model31.Cost = Objective(expr =1200*model31.x1+2580*model31.x2 , sense=minimize)

In [41]:
#declare costraints
model31.constr1= Constraint(expr=model31.x1+model31.x2>=145)
model31.constr2=Constraint(expr=2.18*model31.x1+1.08*model31.x2<=980)
model31.constr3=Constraint(expr=model31.x1>=147)
model31.constr4=Constraint(expr=model31.x2>=222)

In [42]:
model31.pprint()


2 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals

1 Objective Declarations
    Cost : Size=1, Index=None, Active=True
        Key  : Active : Sense    : Expression
        None :   True : minimize : 1200*x1 + 2580*x2

4 Constraint Declarations
    constr1 : Size=1, Index=None, Active=True
        Key  : Lower : Body    : Upper : Active
        None : 145.0 : x1 + x2 :  +Inf :   True
    constr2 : Size=1, Index=None, Active=True
        Key  : Lower : Body              : Upper : Active
        None :  -Inf : 2.18*x1 + 1.08*x2 : 980.0 :   True
    constr3 : Size=1, Index=None, Active=True
        Key  : Lower : Body : Upper : Active
        None : 147.0 :   x1 :  +Inf :   True
    constr4 : Size=1, Index=None, Active=T

In [43]:
!apt-get install -y -qq glpk-utils

In [45]:
SolverFactory('glpk', executable='/usr/bin/glpsol').solve(model31).write()

# ==========================================================
# = Solver Results                                         =
# ==========================================================
# ----------------------------------------------------------
#   Problem Information
# ----------------------------------------------------------
Problem: 
- Name: unknown
  Lower bound: 749160.0
  Upper bound: 749160.0
  Number of objectives: 1
  Number of constraints: 5
  Number of variables: 3
  Number of nonzeros: 7
  Sense: minimize
# ----------------------------------------------------------
#   Solver Information
# ----------------------------------------------------------
Solver: 
- Status: ok
  Termination condition: optimal
  Statistics: 
    Branch and bound: 
      Number of bounded subproblems: 0
      Number of created subproblems: 0
  Error rc: 0
  Time: 0.017976045608520508
# ----------------------------------------------------------
#   Solution Information
# ------------------------------

In [46]:
#display solution
print('\nCost=',model31.Cost())
print('decision Variables')
print('x1=', model31.x1())
print('x2=',model31.x2())
print('Constraints')
print('constraint 1=',model31.constr1())
print('constraint2=',model31.constr2())
print('constraint3=',model31.constr3())
print('constraint4=',model31.constr4())



Cost= 749160.0
decision Variables
x1= 147.0
x2= 222.0
Constraints
constraint 1= 369.0
constraint2= 560.22
constraint3= 147.0
constraint4= 222.0


 total cost: 749160.0
 \\
    decision variable:
   x1 =  147.0,
x2 =  222.0 \\
   Activeness  constraints: \\
   constraint 1 not active  LHS= x1+x2 = 147+222 =369 because is not equal  RHS. \\
   Constraint 2  not active se LHS = 6731.52 because is not equal to RHS \\
   constraint 3 is active  LHS = 147 because is equal to RHS \\
   constraint 4 is active  LHS = 222 because is equal to RHS \\
   after  order jen industries has  increase there productity two  product P and Q.
    optimal cost  Insdustries has to produce 147 kgs of the pruduct P and 222 kgs of the product Q.